EIPS/eip-1193.md

```diff
@@ -7,104 +7,183 @@ status: Draft
-```
+- when the Provider first connects to a chain after being initialized
+- when the Provider connects to a chain, after the `close` event was emitted
 
-You can detect which chain by sending `eth_chainId`:
+```typescript
+interface ProviderConnectInfo {
+  chainId: string;
+  [key: string]: unknown;
+}
 
-```js
-const chainId = await ethereum.send('eth_chainId');
-> 1
+ethereum.on('connect', listener: (connectInfo: ProviderConnectInfo) => void): ethereum;
 ```
 
+The event emits an object with a hexadecimal string `chainId` per the `eth_chainId` Ethereum RPC method, and other properties as determined by the Provider.
+
 #### close
 
-The provider emits `close` on disconnect from a network.
+The Provider emits `close` when it becomes disconnected from all chains.
 
-```js
-ethereum.on('close', listener: (code: Number, reason: String) => void): this;
+```typescript
+ethereum.on('close', listener: (code: number, reason: string) => void): ethereum;
 ```
 
-The event emits with `code` and `reason`. The code follows the table of [`CloseEvent` status codes](https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent#Status_codes).
+This event emits with `code` and `reason`. The code follows the table of [`CloseEvent` status codes](https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent#Status_codes).
 
 #### chainChanged
 
-The provider emits `chainChanged` on connect to a new chain.
+The Provider emits `chainChanged` when connecting to a new chain.
 
-```js
-ethereum.on('chainChanged', listener: (chainId: Integer) => void): this;
+```typescript
+ethereum.on('chainChanged', listener: (chainId: string) => void): ethereum;
 ```
 
-The event emits with integer `chainId`, the new chain returned from `eth_chainId`.
+The event emits a hexadecimal string `chainId` per the `eth_chainId` Ethereum RPC method.
+
+#### networkChanged (DEPRECATED)
 
-The event `networkChanged` is deprecated in favor of `chainChanged`. For more info, see [EIP 155: Simple replay attack protection](https://eips.ethereum.org/EIPS/eip-155) and [EIP 695: Create eth_chainId method for JSON-RPC](https://eips.ethereum.org/EIPS/eip-695).
+The event `networkChanged` is deprecated in favor of `chainChanged`. For details, see [EIP 155: Simple replay attack protection](https://eips.ethereum.org/EIPS/eip-155) and [EIP 695: Create eth_chainId method for JSON-RPC](https://eips.ethereum.org/EIPS/eip-695).
 
 #### accountsChanged
 
-The provider emits `accountsChanged` if the accounts returned from the provider (`eth_accounts`) changes.
+The Provider emits `accountsChanged` if the accounts returned from the Provider (`eth_accounts`) change.
 
-```js
-ethereum.on('accountsChanged', listener: (accounts: Array<String>) => void): this;
+```typescript
+ethereum.on('accountsChanged', listener: (accounts: Array<string>) => void): ethereum;
 ```
 
-The event emits with `accounts`, an array of the accounts' addresses.
+The event emits with `accounts`, an array of account addresses.
```

Comment:
Should this specify whether or not the addresses are `0x`-prefixed and/or EIP-55 checksum-style strings?

Reply:
I clarified that it must match the return value of `eth_accounts`.